Source/UZKFileInfo.m

```diff
@@ -65,6 +97,24 @@ - (NSDate *)timestamp {
 
 
 
+#pragma mark - Private Class Methods
+
+
++ (BOOL)itemIsDirectory:(unz_file_info64 *)fileInfo {
+    UZKZipOS zipOSClassMapping = fileInfo->version >> 8;
+    if (zipOSClassMapping == UZKZipOSMSDOS || zipOSClassMapping == UZKZipOSWindowsNT) {
+        return 0x01 == (fileInfo->external_fa >> 4) && ![UZKFileInfo itemIsSymbolicLink:fileInfo];
+    }
+    
+    return S_IFDIR == (S_IFMT & fileInfo->external_fa >> 16) && ![UZKFileInfo itemIsSymbolicLink:fileInfo];
+}
+
++ (BOOL)itemIsSymbolicLink:(unz_file_info64 *)fileInfo {
+    return S_IFLNK == (S_IFMT &fileInfo->external_fa >> 16);
```

Comment:
Same as above, and also please add a space between `&` and `fileInfo` - it looks like you're passing its address, rather than applying a bitwise AND.